inc/parser/xhtml.php

```diff
@@ -62,75 +65,80 @@ class Doku_Renderer_xhtml extends Doku_Renderer {
      *
      * @author Adrian Lang <lang@cosmocode.de>
      */
-    public function startSectionEdit($start, $data) {
+    public function startSectionEdit($start, $data)
+    {
         if (!is_array($data)) {
             msg(
                 sprintf(
                     'startSectionEdit: $data "%s" is NOT an array! One of your plugins needs an update.',
-                    hsc((string) $data)
+                    hsc((string)$data)
                 ), -1
             );
 
             // @deprecated 2018-04-14, backward compatibility
             $args = func_get_args();
-            $data = array();
-            if(isset($args[1])) $data['target'] = $args[1];
-            if(isset($args[2])) $data['name'] = $args[2];
-            if(isset($args[3])) $data['hid'] = $args[3];
+            $data = [];
+            if (isset($args[1])) $data['target'] = $args[1];
+            if (isset($args[2])) $data['name'] = $args[2];
+            if (isset($args[3])) $data['hid'] = $args[3];
         }
         $data['secid'] = ++$this->lastsecid;
         $data['start'] = $start;
         $this->sectionedits[] = $data;
-        return 'sectionedit'.$data['secid'];
+        return 'sectionedit' . $data['secid'];
     }
 
     /**
      * Finish an edit section range
      *
-     * @param int  $end     The byte position for the edit end; null for the rest of the page
+     * @param int $end The byte position for the edit end; null for the rest of the page
      *
      * @author Adrian Lang <lang@cosmocode.de>
      */
-    public function finishSectionEdit($end = null, $hid = null) {
-        if(count($this->sectionedits) == 0) {
+    public function finishSectionEdit($end = null, $hid = null)
+    {
+        if (count($this->sectionedits) == 0) {
             return;
         }
         $data = array_pop($this->sectionedits);
-        if(!is_null($end) && $end <= $data['start']) {
+        if (!is_null($end) && $end <= $data['start']) {
             return;
         }
-        if(!is_null($hid)) {
+        if (!is_null($hid)) {
             $data['hid'] .= $hid;
         }
-        $data['range'] = $data['start'].'-'.(is_null($end) ? '' : $end);
+        $data['range'] = $data['start'] . '-' . (is_null($end) ? '' : $end);
         unset($data['start']);
-        $this->doc .= '<!-- EDIT'.hsc(json_encode ($data)).' -->';
+        $this->doc .= '<!-- EDIT' . hsc(json_encode($data, JSON_THROW_ON_ERROR)) . ' -->';
```

Comment:
Since $data is created above this should never throw. If it does something is seriously wrong and I'm fine with having it bubble up all the way to the fatal error handler.